src/parser/params.rs

```diff
@@ -8,23 +8,54 @@ use crate::{
 
 use super::{precedence::Precedence, ParseResult, Parser};
 
+#[derive(Debug)]
+pub enum ParamPosition {
+    Function,
+    Method(String),
+    AbstractMethod(String),
+}
+
 impl Parser {
-    pub(crate) fn param_list(&mut self) -> Result<ParamList, ParseError> {
+    pub(crate) fn param_list(&mut self, position: ParamPosition) -> Result<ParamList, ParseError> {
         let mut params = ParamList::new();
 
         while !self.is_eof() && self.current.kind != TokenKind::RightParen {
             let mut param_type = None;
 
             let mut flags: Vec<PropertyFlag> = Vec::new();
 
-            // FIXME: You should only be able to have a combination of 1 visibility + readonly modifier.
-            while matches!(
+            if matches!(
```

Comment:
This code won't support `readonly public` will it? That's totally valid PHP, there's no language restriction on what order those flags/modifiers go in.

Reply:
oh, nice catch, will fix it :)

Reply:
fixed!